### PR TITLE
Added missing XML documentation and resolved some discrepancies in existing comments

### PR DESCRIPTION
**Pull request by rabadashTheFool**
Added missing XML documentation and resolved some discrepancies in existing comments

WE2-1008

Signed-off-by: Mihkel Kivisild <mihkel.kivisild@cgi.com>

### CODE REVIEW

File: src/WebEid.Security/Validator/AuthTokenSignatureValidator.cs [hunk 1]
```diff
@@ -47,17 +50,27 @@ public class AuthTokenSignatureValidator
...
+        /// <param name="tokenSignature">The base64-encoded signature from the token.</param>
+        /// <param name="currentNonce">The current nonce value.</param>
         public void Validate(string algorithm, SecurityKey publicKey, string tokenSignature, string currentNonce)
         {
             if (string.IsNullOrEmpty(currentNonce))
             { throw new ArgumentNullException(nameof(currentNonce)); }
             if (publicKey == null)
             { throw new ArgumentNullException(nameof(publicKey)); }
             RequireNotEmpty(algorithm, nameof(algorithm));
-            RequireNotEmpty(tokenSignature, nameof(tokenSignature));
```
[the remark this answers is not in the document]
Comment (rabadashTheFool) [reply src/WebEid.Security/Validator/AuthTokenSignatureValidator.cs]: Yes, this was a mistake. I have fixed it.

File: src/WebEid.Security/Validator/AuthTokenValidationConfiguration.cs [hunk 2]
```diff
@@ -46,19 +49,38 @@ private AuthTokenValidationConfiguration(AuthTokenValidationConfiguration other)
...
+        /// A value indicating whether user certificate revocation check with OCSP is enabled.
+        /// By default the revocation check is enabled.
+        /// </summary>
         public bool IsUserCertificateRevocationCheckWithOcspEnabled { get; set; } = true;
 
+        /// <summary>
+        /// The OCSP request timeout.
+        /// </summary>
         public TimeSpan OcspRequestTimeout { get; set; } = TimeSpan.FromSeconds(5);
         public TimeSpan AllowedOcspResponseTimeSkew { get; set; } = TimeSpan.FromMinutes(15);
```
[the remark this answers is not in the document]
Comment (rabadashTheFool) [reply src/WebEid.Security/Validator/AuthTokenValidationConfiguration.cs:76]: Added missing documentation.

File: src/WebEid.Security/Validator/Ocsp/Service/IOcspService.cs [hunk 3]
```diff
@@ -23,11 +23,26 @@ namespace WebEid.Security.Validator.Ocsp.Service
...
+        /// <summary>
+        /// Gets the access location (URI) of the OCSP service.
+        /// </summary>
         Uri AccessLocation { get; }
 
+        /// <summary>
+        /// Validates the responder certificate and its produced timestamp.
+        /// </summary>
+        /// <param name="responderCertificate">The responder's X.509 certificate.</param>
+        /// <param name="producedAt">The timestamp when the OCSP response was produced.</param>
```
[the remark this answers is not in the document]
Comment (rabadashTheFool) [reply src/WebEid.Security/Validator/Ocsp/Service/IOcspService.cs]: Updated the documentation

**Review by mrts — APPROVED** [review]

**Comment by rabadashTheFool** [comment]
I updated the targetframwork for Tests project to .Net 8 as per comments under WE2-1008.

**Review by metsma — APPROVED** [review]

**Pull request merged**